src/containers/my-certificates/certificate-table/certificate-table.js

```diff
@@ -1,20 +1,28 @@
-import React from 'react';
-
+import React, { useMemo } from 'react';
 import { useTranslation } from 'react-i18next';
+
 import { Table, TableBody, TableCell, TableHead, TableRow } from '@material-ui/core';
 import { useStyles } from './certificate-table.styles';
-
+import { ROW_FIELDS } from '../../../configs/index';
 import CertificateItem from '../cetrificate-item';
 
 const CertificateTable = ({ items }) => {
+  const certificatesArr = items;
```

Comment:
You are making not a copy but a link for items. And still mutating in memorizedItems. You don't need this const. You can make all these logic in useMemo.

Reply:
Can I do such way ? 


—————BEFORE———— 

const CertificateTable = ({ items }) => {
  const certificatesArr = items;
  const styles = useStyles();
  const { t } = useTranslation();

  const memoizedItems = useMemo(
    () => certificatesArr.sort((a, b) => new Date(b.dateStart) - new Date(a.dateStart)),
    [certificatesArr]
  );

  const certificateItems = memoizedItems.map((item) => (
    <CertificateItem item={item} key={item._id} />
  ));

  const headerItems = ROW_FIELDS.map((item) => (
    <TableCell key={item._id}>{t(`certificate.${item}`)}</TableCell>
  ));

  return (…);
};

export default CertificateTable;

—————AFTER———— 

const CertificateTable = ({ items }) => {
  const styles = useStyles();
  const { t } = useTranslation();

  const certificateItems = useMemo(() => {
    return items
      .sort((a, b) => new Date(b.dateStart) - new Date(a.dateStart))
      .map((item) => <CertificateItem item={item} key={item._id} />);
  }, [items]);

  const headerItems = useMemo(() => {
    return ROW_FIELDS.map((item) => (
      <TableCell key={item._id}>{t(`certificate.${item}`)}</TableCell>
    ));
  }, [ROW_FIELDS]);

  return (…);
};

export default CertificateTable;